components/constellation/constellation.rs

```diff
@@ -1930,10 +1936,19 @@ impl<Message, LTF, STF> Constellation<Message, LTF, STF>
             // Add new pipeline to navigation frame, and return frames evicted from history.
             self.pipelines.get_mut(&frame_change.new_pipeline_id)
                           .map(|pipeline| pipeline.frame = Some(frame_id));
-            self.frames.get_mut(&frame_id).map(|frame| frame.load(frame_change.new_pipeline_id));
+            if frame_change.replace {
+                let entry = self.frames.get_mut(&frame_id).map(|frame| {
+                    frame.replace_current(frame_change.new_pipeline_id)
+                });
+                if let Some(entry) = entry {
```

Comment:
I should probably warn, the `None` case will only happen if the frame lookup fails.